sematic/plugins/publishing/slack.py

```diff
@@ -0,0 +1,114 @@
+# Standard Library
+import logging
+from typing import Type
+
+# Third-party
+import requests
+
+# Sematic
+from sematic.abstract_plugin import (
+    AbstractPlugin,
+    AbstractPluginSettingsVar,
+    PluginVersion,
+)
+from sematic.config.settings import get_plugin_setting
+from sematic.db.models.resolution import Resolution, ResolutionStatus
+from sematic.db.queries import get_run
+from sematic.plugins.abstract_publisher import AbstractPublisher
+from sematic.utils.memoized_property import memoized_property
+
+logger = logging.getLogger(__name__)
+
+_PLUGIN_VERSION = (0, 1, 0)
+_SLACK_HEADERS = {"Content-Type": "application/json"}
+_SLACK_URL_TEMPLATE = "https://hooks.slack.com/services/{webhook_token}"
+# TODO: implement server-side gnostic components that build urls that point to specific
+#  resources - e.g. have resolutions.py build this url for us
+# currently, this needs to be updated here whenever the endpoints change
```

Comment:
I needed this in the upload work. Instead I returned relative URLs and made the client construct it. Ugly.

Maybe `SEMATIC_URL` is a `ServerSettings` setting instead of `SlackPublisher` setting, and we can use it anywhere we need to generate absolute URLs.

I think I would prefer that so that it would make sense to rename `SLACK_EXTERNAL_URL` to `SEMATIC_URL`.

Reply:
Renamed to `SEMATIC_DASHBOARD_URL`, as a Server setting.